src/components/tx/SignOrExecuteForm/SignForm.tsx

```diff
@@ -101,6 +101,8 @@ export const SignForm = ({
   const submitDisabled =
     !safeTx || !isSubmittable || disableSubmit || cannotPropose || (needsRiskConfirmation && !isRiskConfirmed)
 
+  const isSafeAppTransaction = !!origin
```

Comment:
I think origin is also non-null in case of WalletConnect txs and probably swaps/staking/bridging as well.

Reply:
I think we should hide it for those transactions as well to be on the safe side. At least for swaps the button is already disabled since it is a delegate call:
<img width="715" alt="Screenshot 2024-12-09 at 10 29 49" src="https://github.com/user-attachments/assets/a1f17435-6aa6-4f0d-b7b9-54a6626f3e0e">